services/mailservers/database.go

```diff
@@ -52,18 +56,18 @@ func (m Mailserver) PeerID() (peer.ID, error) {
 		return "", errors.New("not available")
 	}
 
-	pID, err := peer.Decode(m.UniqueID())
+	addr, err := multiaddr.NewMultiaddr(m.Address)
 	if err != nil {
 		return "", err
 	}
 
-	return pID, nil
+	return utils.GetPeerID(addr)
 }
 
 func (m Mailserver) UniqueID() string {
 	if m.Version == 2 {
-		s := strings.Split(m.Address, "/")
-		return s[len(s)-1]
+		p, _ := m.PeerID()
```

Comment:
Ignoring the error here is ugly, but it's getting fixed in the mailserver refactoring I'm doing to avoid using only multiaddresses. Still, it's not like the previous version was better since it assumed that the multiaddresses would always have a peerID anyways